MetaMask/metamask-extension
# Refactor usages of conversion util in the app folder and shared folder in favor of Numeric

## Explanation
In #17324 we added a module for number conversions that is easier to understand and better documented, and covered by lots of tests. This PR aims to refactor usages of the old conversionUtil method in the app/ and shared/ folder such that it can eventually be removed from the codebase in a future PR.  Note that not all method changes here are changing usage of conversionUtil, some are just changed to use other, approved helper methods that utilize the new Numeric module under the hood.

## Manual Testing Steps
- The only real manual testing steps would be to use the application and make sure that number representations add up.
- Obviously check for any .test file changes as I shouldn't have had to modify test files to make these work if there are no functional changes. The one caveat here is that I moved a test for the 'calcGasTotal' method from the send slice to the place where it actually matters. We were mocking everything about that method and testing that our mocks worked which is kind of weird. The new test actually tests the method itself and not a mock.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
